pkg/agent/controller/egress/egress_controller.go

```diff
@@ -66,7 +67,8 @@ const (
 	egressIPIndex       = "egressIP"
 	externalIPPoolIndex = "egressExternalIPPool"
 
-	DefaultEgressRunDir = "/var/run/antrea/egress"
+	// egressDummyDevice is the dummy device that holds the Egress IPs configured to the system by antrea-agent.
+	egressDummyDevice = "antrea-egress0"
```

Comment:
Could we reuse an existing one like antrea-gw0, or br-int?

Reply:
If we assign to antrea-gw0 which already has its own IP or br-int which may hold the IP moved from uplink device in the future, the code that manages those interfaces' IPs may be affected because they need to be aware of such dummy external IPs, for example currently agent has [a goroutine](https://github.com/antrea-io/antrea/blob/c21592c0e718539f6a190843eb9fff9af8d3d175/pkg/agent/agent.go#L886) to ensure antrea-gw0 has only the desired IP gateway IPs. And the IPAssigner needs to know the gateway IPs the other logic want to keep, which is not clearer than a separate dummy interface. 
In the future, new ipAssigner with its own dummy interface can be used by other features that need to assign dummy IPs, making them use their own interfaces may avoid multiple controllers managing same resource for ease of cleanup and troubleshooting. For example, it would like this: antrea-egress0 holds the egress IPs, antrea-ingress0 holds the ingress IPs, antrea-gw0 holds the gateway IPs.

Reply:
Got your point. Question - will ingress IP assignment be different from egress, so you will need two interfaces?

Reply:
In my mind ingress IP assignment can reuse the code and initialize another IPAssigner with its own dummy device "antrea-ingress0". They can also share a device but it will require a little interaction between the two modules so that they will not clean up other's IPs, but it wouldn't be too difficult. I'm just not sure how will it be implemented yet. If you prefer using single device for all external IPs, maybe I can rename it to "antrea-external0"?

Reply:
Yes, I am trying to understand this. Do you think we need a separate IPAssigner for ingress IPs, why?

Reply:
For ease of cleanup, otherwise we need to collect both desired ingress IPs and desired egress IPs somewhere and do the cleanup. https://github.com/antrea-io/antrea/blob/f4994b34d283c54a4b3c8c829f799ee47658c455/pkg/agent/controller/egress/egress_controller.go#L293-L298


And troubleshooting, it's more clear that which IPs are assigned to the system for which purpose by viewing `ip addr`

Reply:
I do not have a strong opinion. I do not know other devices use two interfaces for ingress/egress IPs, but maybe that is not a big deal.

Reply:
Since this is alpha, I think we still have a chance to rename it if we want to share a device with ingress IPs later. And it shouldn't be hard to rename it gracefully if we want to. I will keep the name for now.